core/src/main/scala/org/broadinstitute/dsde/rawls/openam/StandardUserInfoDirectives.scala

```diff
@@ -28,7 +28,7 @@ trait StandardUserInfoDirectives extends UserInfoDirectives {
     case (token, userId, expiresIn, email, googleTokenOpt) => {
       val userInfo = UserInfo(RawlsUserEmail(email), OAuth2BearerToken(token), expiresIn.toLong, RawlsUserSubjectId(userId), googleTokenOpt.map(OAuth2BearerToken))
       onSuccess(getWorkbenchUserEmailId(userInfo).map {
-        case Some(petOwnerUser) => userInfo.copy(userEmail = petOwnerUser.userEmail, userSubjectId = petOwnerUser.userSubjectId)
+        case Some(petOwnerUser) => userInfo.copy(userEmail = petOwnerUser.userEmail)
```

Comment:
This code was clobbering the auth header's subject ID the sam userSubjectId field in the response payload which is the Sam user ID. This was incorrect; code in rawls expects this to be a cloud provider's subject ID (i.e., google or b2c). I'm removing the set of the subject ID here and leaving the subject ID from the header as expressed on L23 above.

Reply:
I'm not quite sure about this change. Copying over `petOwnerUser.userSubjectId` and `petOwnerUser.userEmail` lets us treat a pet service account as the pet's owner. If we only copy over the `userEmail`, we'll be passing around a `UserInfo` that has the pet SA's Google subject id, but the owning user's email.
 
From what I can tell, it seems like the change we really need is to surface the pet's owner's cloud provider subject ID in the Sam response so that we can copy _that_ over into this `UserInfo` object and treat a pet as its owner.

Reply:
I agree with Marc. Leaving it as the pet SA's google subject id will be incorrect for notifications and auditing.

Reply:
Sensible, I'll look at the Sam endpoint, thanks both.